shared/chat/conversation/input-area/normal/index.js

```diff
@@ -1,23 +1,89 @@
 // @flow
 import * as React from 'react'
-import {Input as TextInput} from '../../../../common-adapters'
-import MentionInput from './mention-input'
+import * as Kb from '../../../../common-adapters'
+import * as Styles from '../../../../styles'
+import * as Constants from '../../../../constants/chat2'
+import {emojiIndex} from 'emoji-mart'
+import PlatformInput from './platform-input'
 import {type InputProps} from './types'
 import {throttle} from 'lodash-es'
 
 // Standalone throttled function to ensure we never accidentally recreate it and break the throttling
 const throttled = throttle((f, param) => f(param), 2000)
 
-class Input extends React.PureComponent<InputProps> {
+const suggestorToMarker = {
+  channels: '#',
+  emoji: ':',
+  users: '@',
+}
+
+const suggestorKeyExtractors = {
+  emoji: (item: {id: string}) => item.id,
+  users: ({username, fullName}: {username: string, fullName: string}) => username,
+}
+
+const emojiDatasource = (filter: string) => (filter.length >= 2 ? emojiIndex.search(filter) : [])
+const emojiRenderer = (item, selected: boolean) => (
+  <Kb.Box2
+    direction="horizontal"
+    fullWidth={true}
+    style={Styles.collapseStyles([
+      styles.suggestionBase,
+      {
+        backgroundColor: selected ? Styles.globalColors.blue4 : Styles.globalColors.white,
+      },
+    ])}
+    gap="small"
+  >
+    <Kb.Emoji emojiName={item.colons} size={24} />
+    <Kb.Text type="BodySmallSemibold">{item.colons}</Kb.Text>
+  </Kb.Box2>
+)
+const emojiTransformer = (
+  emoji: {colons: string, native: string},
+  {position: {end, start}, text},
+  preview
+) => {
+  const toInsert = Styles.isMobile ? emoji.native : emoji.colons
```

Comment:
this pattern is repeated 3x, so it's probably worth pulling out into its own helper function